extensions/resteasy-reactive/rest-client/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/devservices/DevServicesRestClientHttpProxy.java

```diff
@@ -0,0 +1,276 @@
+                    log.debug("Unable to determine uri or url for config key '" + configKey + "'");
+                    break;
+                }
+                producer.produce(new RestClientHttpProxyBuildItem(matchingBI.getClassInfo().name().toString(), baseUri.get()));
+            } else {
+                // now we check if the configKey was actually a class name
+                ClassInfo classInfo = index.getClassByName(configKey);
+                if (classInfo == null) {
+                    log.debug(
+                            "Key '" + configKey + "' could not be matched to either a class name or a REST Client's configKey");
+                    break;
+                }
+                Optional<String> baseUri = oneOf(
+                        RestClientConfig.getConfigValue(configKey, "uri", String.class),
+                        RestClientConfig.getConfigValue(configKey, "url", String.class));
+                if (baseUri.isEmpty()) {
+                    log.debug("Unable to determine uri or url for config key '" + configKey + "'");
+                    break;
+                }
+                producer.produce(new RestClientHttpProxyBuildItem(classInfo.name().toString(), baseUri.get()));
+            }
+        }
+    }
+
+    private String sanitizeKey(String key) {
+        if (key.startsWith("\"") && key.endsWith("\"")) {
+            return key.substring(1, key.length() - 1);
+        }
+        return key;
+    }
+
+    @BuildStep
+    public void start(List<RestClientHttpProxyBuildItem> restClientHttpProxyBuildItems,
+            BuildProducer<DevServicesResultBuildItem> devServicePropertiesProducer,
+            CuratedApplicationShutdownBuildItem closeBuildItem) {
+        if (restClientHttpProxyBuildItems.isEmpty()) {
+            return;
+        }
+
+        Set<RestClientHttpProxyBuildItem> currentBIs = new HashSet<>(restClientHttpProxyBuildItems);
+
+        Set<RestClientHttpProxyBuildItem> inRunningAndNotCurrent = new HashSet<>(runningBIs);
+        inRunningAndNotCurrent.removeAll(currentBIs);
+
+        // we need to remove the running ones that should no longer be running
+        for (var running : inRunningAndNotCurrent) {
+            try {
+                log.debug("Attempting to close HTTP proxy server for REST Client '" + running.getClassName() + "'");
+                running.getCloseable().close();
+                log.debug("Closed HTTP proxy server for REST Client '" + running.getClassName() + "'");
+            } catch (IOException e) {
+                throw new UncheckedException(e);
+            }
+        }
+        runningBIs.removeAll(inRunningAndNotCurrent);
+
+        Set<RestClientHttpProxyBuildItem> inCurrentAndNotRunning = new HashSet<>(currentBIs);
+        inCurrentAndNotRunning.removeAll(runningBIs);
+
+        if (vertx.get() == null) {
+            // TODO: what settings do we need here in order to minimize the footprint?
+            vertx.set(Vertx.vertx());
```

Comment:
Default should not be too bad.

I would recommend:

- disabling async DNS
- reducing the number of event loop / worker thread pool size
- making sure that metrics are disabled

Reply:
I added those along some more settings we have elsewhere